addons/website_sale_comparison/static/src/js/website_sale_comparison.js

```diff
@@ -105,7 +105,8 @@ var ProductComparison = publicWidget.Widget.extend(VariantMixin, {
                 $elem.closest('form').find('.product_template_id').val(),
                 false
             ).then(function (productId) {
-                productId = parseInt(productId, 10);
+                var defaultProductId = parseInt($elem.data('product-product-id'), 10)
+                productId = parseInt(productId, 10) || defaultProductId;
                 if (!productId) {
```

Comment:
can we have !productId now ? 

It is strange to compute the fallback value for a rare case, before to know if need or not...

Reply:
You are right, do you have another idea ?